server/controller/src/main/java/ai/starwhale/mlops/configuration/SlowSqlLogInterceptor.java

```diff
@@ -0,0 +1,153 @@
+        @Signature(type = StatementHandler.class, method = "update", args = {Statement.class})
+})
+@ConditionalOnProperty(prefix = "sw.sql.log-slow-sql", name = "enable", havingValue = "true", matchIfMissing = true)
+public class SlowSqlLogInterceptor implements Interceptor {
+
+    private Configuration configuration = null;
+    private static final ThreadLocal<SimpleDateFormat> DATE_FORMAT_THREAD_LOCAL =
+            ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS"));
+
+    private final int slowSqlMillis;
+
+    private final int maxSqlLength;
+
+    public SlowSqlLogInterceptor(
+            @Value("${sw.sql.slow-sql-millis:100}") int slowSqlMillis,
+            @Value("${sw.sql.max-print-length:200}") int maxSqlLength
+    ) {
+        this.slowSqlMillis = slowSqlMillis;
+        this.maxSqlLength = maxSqlLength;
+    }
+
+
+    @Override
+    public Object intercept(Invocation invocation) throws Throwable {
+        Object target = invocation.getTarget();
+        long startTime = System.currentTimeMillis();
+        StatementHandler statementHandler = (StatementHandler) target;
+        try {
+            return invocation.proceed();
+        } finally {
+            long endTime = System.currentTimeMillis();
+            long cost = endTime - startTime;
+            if (cost >= slowSqlMillis) {
+                BoundSql boundSql = statementHandler.getBoundSql();
+
+                if (configuration == null) {
+                    var parameterHandler = (DefaultParameterHandler) statementHandler.getParameterHandler();
+                    var configurationField = ReflectionUtils.findField(parameterHandler.getClass(), "configuration");
+                    if (configurationField != null) {
+                        ReflectionUtils.makeAccessible(configurationField);
+                        this.configuration = (Configuration) configurationField.get(parameterHandler);
+                    }
+                }
+                if (configuration != null) {
+                    var sql = formatSql(boundSql, configuration);
+                    log.info("Execute SQL：[ {} ] cost[ {} ms]", sql, cost);
+                }
+            }
+        }
+    }
+
+    /**
+     * Get param list
+     *
+     * @param configuration the configuration
+     * @param boundSql      the bound sql
+     *
+     * @return the param list
+     */
+    public String formatSql(BoundSql boundSql, Configuration configuration) {
+        String sql = boundSql.getSql();
+        List<ParameterMapping> parameterMappings = boundSql.getParameterMappings();
+        Object parameterObject = boundSql.getParameterObject();
+
+        if (sql == null || sql.length() == 0) {
+            return "";
+        }
+        if (configuration == null) {
+            return "";
+        }
+        TypeHandlerRegistry typeHandlerRegistry = configuration.getTypeHandlerRegistry();
+        sql = sql.replaceAll("[\\s\n ]+", " ");
```

Comment:
`\n` is a subset of `\s`,  isn't `\s+`  sufficient

Reply:
fixed